### DISCUSSION TITLE
Is it possible to refresh KafkaJaasLoginModuleInitializer bean and therefore the JAAS configuration of a KafkaClient?

### DISCUSSION THREAD

**Discussion by tunix** [spring-projects/spring-kafka#2217] [Q&A]
Hi,

I have a project which connects to a secure Kafka using SCRAM user credentials. Instead of hardcoding user password into the application.yaml, I'm using a more dynamic approach where I get the password from a vault service. This mechanism checks for changes to the password every N seconds and updates properties (+ executes callbacks) if needed.

If I refresh the `KafkaJaasLoginModuleInitializer` bean after a password change, how does it affect the KafkaClient? I tried to read through the source code but I'm not sure whether this login procedure only takes place during initialization of connections or elsewhere.

**Comment by garyrussell** [comment]
The initializer simply configures the global static `javax.security.auth.login.Configuration` for the JVM.

We don't know the client internals intimately here; you might needs to set the config via the consumer/producer properties instead.

https://kafka.apache.org/documentation/#configuration

I see properties about sasl renewal there; but you'd be better asking a wider audience on Stack Overflow on on one of the Apache Kafka Mailing lists.
